src/FakeItEasy/ExceptionThrowerConfigurationExtensions.cs

```diff
@@ -130,6 +134,11 @@ public static IAfterCallSpecifiedConfiguration Throws<T1>(this IExceptionThrower
                 });
         }
 
+#if DISABLED
+
+        // This method already has a generic parameter, and it can't be inferred; if we add another, we make it unusable,
+        // especially since the new one is something unintelligible like IAnyCallConfigurationWithReturnTypeSpecificied<Foo>...
+

```

Comment:
> If the interfaces were abstract classes would you feel better about it?

I guess so; especially since only one of the methods would need to be abstract. But it's not really an option for FIE, because RuleBuilder et al. implement multiple configuration interfaces, and .NET only supports one base class.

Anyway, having the extension methods declared in the interfaces is probably the best option here. What bothers me the most is the repetition. For instance, 3 classes implement `IOutAndRefParametersConfiguration.AssignsOutAndRefParametersLazily`, and there are 5 related helper extension methods; these 5 methods will turn into 15 when implemented in each of the configuration classes, with almost exactly the same code. Maybe we could keep the extension methods, but make them internal, and use them to implement the interface methods...